### PR TITLE
20230520 intent interface

### PR DESCRIPTION
**Pull request by jorgeantonio21**
Adds `Interface` trait, and derives it for a swap use case (`SwapIntent`).

### CODE REVIEW

File: Cargo.toml [hunk 1]
```diff
@@ -1,6 +1,6 @@
 [workspace]
 members = [
+    "conversions",
```
Comment (benjaminbollen) [inline Cargo.toml]: im confused, didnt we change this previously from "conversions" to "types"?

File: conversions/src/types.rs [hunk 2]
```diff
@@ -165,6 +165,10 @@ impl Signature {
             s: Secp256K1Scalar::from_noncanonical_biguint(BigUint::from_bytes_be(&s)),
         }
     }
+
+    pub fn sign_message(private_key: &PrivateKey, message: &Message) -> Self {
```
Comment (benjaminbollen) [inline conversions/src/types.rs]: should `sign_message` not be a method on `PrivateKey` so that we don't need to expose the private key outside its type?

Comment (jorgeantonio21) [reply conversions/src/types.rs]: Good catch, changed it accordingly

File: Cargo.toml (hunk 1 above)
Comment (jorgeantonio21) [reply Cargo.toml]: We might, I like "conversions", as this library is mostly concerned with converting between `libsecp256k1` and `plonky2_ecdsa` types

**Review by jorgeantonio21 — COMMENTED** [review]
Addressed the `signature` method (added to `PrivateKey`). Happy to change `conversions` naming crate to `types`.

**Review by benjaminbollen — APPROVED** [review]
LGTM

**Pull request merged**